src/btree/bt_sync.c

```diff
@@ -221,8 +227,46 @@ __sync_file(WT_SESSION_IMPL *session, WT_CACHE_OP syncop)
 				leaf_bytes += page->memory_footprint;
 				++leaf_pages;
 			}
+			++page_counter;
+
 			WT_ERR(__wt_reconcile(
 			    session, walk, NULL, WT_CHECKPOINTING, NULL));
+
+			if (timer && page_counter > 5000) {
+				__wt_epoch(session, &cur_time);
+				time_diff = WT_TIMEDIFF_SEC(cur_time,
+						conn->ckpt_tree_verb_time);
+
+				if (time_diff > 20) {
+
+					int_bytes_svpt =
+					    internal_bytes - int_bytes_svpt;
+					int_page_svpt =
+					    internal_pages - int_page_svpt;
+					leaf_bytes_svpt =
+					    leaf_bytes - leaf_bytes_svpt;
+					leaf_svpt = leaf_pages - leaf_svpt;
+
+					__wt_verbose(session,
+					    WT_VERB_CHECKPOINT,
+					    "Checkpoint has been running and"
+					    " took %" PRIu64 "sec to write: %"
+					    PRIu64" leaf pages (%" PRIu64 "B),"
+					    "%" PRIu64 " internal pages (%"
+					    PRIu64 "B)", time_diff, leaf_svpt,
+					    leaf_bytes_svpt, int_page_svpt,
+					    int_bytes_svpt);
+
+					int_bytes_svpt = internal_bytes;
+					int_page_svpt = internal_pages;
+					leaf_bytes_svpt = leaf_bytes;
+					leaf_svpt = leaf_pages;
+					conn->ckpt_tree_verb_time = cur_time;
```

Comment:
I don't like the idea of having ckpt_tree_verb_time initialized somewhere else. I understand what you are trying to do, but makes me uneasy and I am unable to figure out what the time diff really is: 

__checkpoint_tree() initialises ckpt_tree_verb_time. Then __sync_file() and __ckpt_process get time locally and subtract ckpt_tree_verb_time to obtain time elapsed since last reset of ckpt_tree_verb_time.

__wt_reconcile in __sync_file ends up calling __ckpt_process which will reset conn->ckpt_tree_verb_time if checkpoint delete is involved? So does that mean "time_diff > 20" is from the time checkpoint was deleted ?

Why not have start_time and end_time locally around delete checkpoint and the for loop in sync_file that iterates through pages are writes them to disk. It will be cleaner to understand. You have looked at this code more than I have, probably I am missing something ?

Reply:
The idea here is to start a timer at the time when we start checkpointing a tree(file), and then  log progress message for the steps that are taking considerable time , while checkpointing that file. Hence the timer is started as we enter the function __checkpoint_tree. 
In the course of checkpointing the tree, we also delete the checkpoints that are no longer required. It was noticed in certain instances that deleting checkpoint was taking considerable time while checkpointing a tree. So it is worth logging a message at this step and then log another message for sync operation it that is taking more than 20 sec after this step.